lib/iris/tests/integration/test_cube.py

```diff
@@ -55,5 +58,20 @@ def test_agg_by_aux_coord(self):
         self.assertEqual(res_cell_methods.method, 'mean')
 
 
+class TestDataFillValue(tests.IrisTest):
+    def test_real(self):
+        data = np.ma.masked_array([1, 2, 3], [0, 1, 0], fill_value=10)
+        cube = Cube(data)
+        cube.data.fill_value = 20
```

Comment:
I think you wanted the set fill value method, no? Otherwise, how can this possibly ever fail (unless the object uses properties to explicitly avoid people being able to set their fill value...)

Reply:
The point is, if `cube.data` is a view, setting the fill value on it doesn't change the array owned by the cube. I'm pretty sure setting the fill value behaves the same as `set_fill_value()`